app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,191 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Nested;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+
+    private static final String TASK_TITLE = "test";
+    private TaskService taskService;
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+    }
+
+    void createTestTask() {
+        Task task = new Task();
+
+        task.setTitle("Test1");
+        taskService.createTask(task);
+
+        task.setTitle("Test2");
+        taskService.createTask(task);
+    }
+
+    @Nested
+    @DisplayName("getTasks 메소드는")
+    class Describe_getTasks {
+
+        @Nested
+        @DisplayName("만약 Task가 아무것도 없다면")
+         class Context_with_nothing {
+            @Test
+            @DisplayName("빈 list를 리턴한다.")
+            void it_return_empty_list() {
+                assertThat(taskService.getTasks()).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 Task가 1개 이상 존재한다면")
+        class Context_with_anything {
+            @Test
+            @DisplayName("전체 Task의 list를 리턴한다.")
+            void it_return_every_task() {
+                createTestTask();
+
+                assertThat(taskService.getTasks()).hasSize(2);
+            }
+        }
```

Comment:
잘작성하셨어요. 그런데 `createTestTask();`는 `Context_with_anything`에서 준비해 주셔야 합니다.
예를 들어 다음과 같이 할 수 있을 거에요. 제가 작성한 한 줄 주석은 설명을 돕기 위해 추가한 것이고, 실제로는 불필요합니다.
```suggestion
        @Nested
        @DisplayName("만약 Task가 1개 이상 존재한다면")
        class Context_with_anything {
            @BeforeEach() {
                // 여기에서 실험을 준비합니다.
                // "만약 Task가 1개 이상 존재한다면" 에 해당하는 코드를 여기에 위치시킵니다.
                createTestTask();
            }

            @Test
            @DisplayName("전체 Task의 list를 리턴한다.")
            void it_return_every_task() {
                // 여기에서 실험을 수행하고 결과를 검증합니다.
                // 전체 Task의 list 가 리턴되는지를 확인하도록 합니다.
                assertThat(taskService.getTasks()).hasSize(2);
            }
        }
```

Reply:
실험 전에 조건을 채워야 하는 부분에서 실행할 메서드 명은 적합하게만 작성하면 되는건가요?
`@BeforeEach
void ready() {....}`

이런 메서드 명으로 생각해봤습니다

Reply:
경우에 따라 열심히 생각해서 지어야 합니다. 규칙은 따로 없어요.